lib/map.dl

```diff
@@ -0,0 +1,150 @@
+/*
+ * Functions for working with maps (see basic map operations defined in
+ * `ddlog_std.dl`).
+ */
+
+/* Applies closure `f` to each value in the map. */
+function map(m: Map<'K,'V1>, f: function('V1): 'V2): Map<'K,'V2> {
+    var res = map_empty();
```

Comment:
how about a constructor with size hint?

Reply:
Maps and sets don't have one. Only vectors do.

Reply:
that seems a misdesign.

Reply:
You cannot easily pre-allocate memory for a map or set as they are backed by trees.  I don't think I've ever seen a library in any language that does that.